lua/src/macro_cache.cc

```diff
@@ -172,7 +243,67 @@ void macro_cache::write(misc::shared_ptr<io::data> const& data) {
  *  @param in  The event.
  */
 void macro_cache::_process_instance(instance_broadcast const& in) {
-  _instances[in.poller_id] = in;
+  unsigned int poller_id(in.poller_id);
+
+  QSet<unsigned int> hosts_removed;
+  for (QHash<unsigned int, neb::host>::iterator
+         it(_hosts.begin()),
+         end(_hosts.end());
+       it != end; ) {
+    if (it->poller_id == poller_id) {
+      hosts_removed << it->host_id;
+      it = _hosts.erase(it);
+    }
+    else
+      ++it;
+  }
+
+  for (QHash<unsigned int, neb::host_group>::iterator
+         it(_host_groups.begin()),
+         end(_host_groups.end());
+       it != end; ) {
+    if (it->poller_id == poller_id)
+      it = _host_groups.erase(it);
+    else
+      ++it;
+  }
+
+  for (QHash<unsigned int,
```

Comment:
In fact it is not possible to remove host groups based on their poller ID. The same host group can span on multiple instances. Unfortunately, I have no solution here.